components/sup/src/http_gateway.rs

```diff
@@ -168,17 +228,24 @@ impl Server {
                 Err(_) => HTTP_THREAD_COUNT,
             };
 
-            let bind = server::new(move || {
+            let server = server::new(move || {
                 let app_state = AppState::new(gateway_state.clone());
-                App::with_state(app_state).configure(routes)
-            }).workers(thread_count)
-            .bind(listen_addr.to_string());
+                App::with_state(app_state)
+                    .middleware(Authentication)
+                    .configure(routes)
+            }).workers(thread_count);
+
+            let config = tls_config(tls_files);
```

Comment:
To increase testability, I think we can keep nearly all the flow logic on this side of the call, and make `tls_config` almost entirely imperative. Combine that with pulling up some of the logic that's currently in the lower block and you get:
```rust
            let bind = match tls_files.map(|(kf, cf)| tls_config(kf, cf)) {
                Some(config_result) => match config_result {
                    Ok(c) => server.bind_rustls(listen_addr.to_string(), c),
                    Err(e) => {
                        error!("Invalid TLS config: {}", e);
                        cvar.notify_one();
                        return;
                    }
                },
                None => server.bind(listen_addr.to_string()),
            };
```

Then you can eliminate the extra scope for the lock by putting the acquisition and assignment all in one statement:
```rust
            *lock.lock().expect("Control mutex is poisoned") = match bind {
                Ok(b) => {
                    b.start();
                    ServerStartup::Started
                }
                Err(e) => {
                    error!("HTTP gateway failed to bind: {}", e);
                    ServerStartup::BindFailed
                }
            };
```

Also, since most values associated with `Result::Err` variants implement `Display`, you can usually use `{}` rather than `{:?}` for nicer formatting.

The necessity of calling `cvar.notify_one()` in two different places is a bit ugly. It might be better to pull the call to `tls_config` one level higher and only start waiting on the condvar once we know we have either a valid TLS configuration or none at all. This function would then take Option<ServerConfig> rather than `Option<(path::PathBuf, path::PathBuf)>`. I'll leave it up to you if you want to go that route.